common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -199,11 +204,15 @@ protected void serviceInit(Configuration externalConf) throws Exception {
     _conf = new Configuration(externalConf);
     URL confOverlayUrl = Thread.currentThread().getContextClassLoader()
         .getResource(SHUFFLE_SERVICE_CONF_OVERLAY_RESOURCE_NAME);
+    String className = YarnShuffleService.class.getName();
     if (confOverlayUrl != null) {
-      logger.info("Initializing Spark YARN shuffle service with configuration overlay from {}",
-          confOverlayUrl);
       _conf.addResource(confOverlayUrl);
     }
+    String logsNamespace = _conf.get(SPARK_SHUFFLE_SERVICE_LOGS_NAMESPACE_KEY, "");
+    String loggerName = logsNamespace.isEmpty() ? className : className + " " + logsNamespace;
```

Comment:
Maybe we can use something slightly more structured than a space

Right now assuming the namespace is "version311" we'll have something like:
```
INFO network.yarn.YarnShuffleService version311 Initialized Spark YARN shuffle ...
```
Maybe this would look better:
```
INFO network.yarn.YarnShuffleService [version311] Initialized Spark YARN shuffle ...
```
Or, since we're considering it part of the name and since frameworks may assume the name looks like a class,
```
INFO network.yarn.YarnShuffleService.version311 Initialized Spark YARN shuffle ...
```
(if we go with the last option because we think following the class-name-like convention is valuable, then we should probably log a warning if the namespace contains spaces)